go/libraries/doltcore/sqle/globalstate/auto_increment_tracker.go

```diff
@@ -38,4 +38,6 @@ type AutoIncrementTracker interface {
 	// below the current value for this table. The table in the provided working set is assumed to already have the value
 	// given, so the new global maximum is computed without regard for its value in that working set.
 	Set(ctx *sql.Context, tableName string, table *doltdb.Table, ws ref.WorkingSetRef, newAutoIncVal uint64) (*doltdb.Table, error)
+
+	AcquireTableLock(ctx *sql.Context, tableName string) (func(), error)
```

Comment:
Give this function some godocs. The name is pretty self explanatory, but generally if something is important enough to be in an interface it merits having godoc. Could be helpful for future code readers to explain the `func()` return param and any rules for using it correctly, for example.

Reply:
Fixed.